### PR TITLE
Modulesync 0.6.0

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

**Comment by bastelfreak** [comment]
@jyaworski @rnelson0 could you take a look please? I'm not sure if I'm using rspec correctly here and the module is missing something or if the test is useless in that case and I should remove it.

**Comment by rnelson0** [comment]
This is tough. You need to either move the checks into where you test `init.pp`, or you need to `include gluster::service` in a `:before` blog or something similar. I don't have it handy right now but I'll look for an example of the latter solution.

**Comment by rnelson0** [comment]
Here you go. You'd have to use `gluster::service` as the class, with whatever arguments it should get, because there's a `require` that wants it, but it's not included by default: https://github.com/puppetlabs/puppetlabs-apache/blob/master/spec/classes/mod/php_spec.rb#L19-L21

This is because `gluster::install` isn't actually meant to be called separately, it's meant to be included as `include gluster` which then includes both. Many modules just have one test file against the `init.pp` file, or if they have a lot of tests, they break them into multiple files that all describe the base class but only test one subclass. Does that make sense?

**Comment by bastelfreak** [comment]
I'm not sure if I understand you correctly. I now added a require for ::gluster::service in the install_spec.rb as a pre_condition. I think that is what you meant?

**Comment by rnelson0** [comment]
Yep. The precondition is like another line in your manifest besides
'include <class name>'., so now both classes end up in the catalog.

-- 

Rob Nelson
rnelson0@gmail.com

File: manifests/service.pp [hunk 1]
```diff
@@ -22,9 +22,9 @@
 # Copyright 2014 CoverMyMeds, unless otherwise noted
 #
 class gluster::service (
-  $ensure = $::gluster::params::service_ensure,
-  $enable = $::gluster::params::service_enable,
-) {
```
Comment (igalic) [inline manifests/service.pp]: so the `::` isn't necessary in variables, but…

File: manifests/service.pp [hunk 2]
```diff
@@ -22,9 +22,9 @@
...
+  $ensure = $gluster::params::service_ensure,
+  $enable = $gluster::params::service_enable,
+) inherits ::gluster::params {
```
Comment (igalic) [inline manifests/service.pp]: it is necessary in class names?

Comment (igalic) [inline manifests/service.pp]: that feels oddly inconsistent.

File: manifests/volume.pp [hunk 3]
```diff
@@ -64,6 +64,8 @@
     } else {
```
Comment (igalic) [inline manifests/volume.pp]: we're failing here, so the else can be left out

File: spec/classes/init_spec.rb [hunk 4]
```diff
@@ -1,91 +1,106 @@
...
           {
-            'data1' => {
-              'replica' => 2,
-              'bricks'  => ['srv1.local:/brick1/brick', 'srv2.local:/brick1/brick'],
-              'options' => ['server.allow-insecure: on'],
+            volumes:
+            {
+              'data1' => {
+                'replica' => 2,
+                'bricks'  => ['srv1.local:/brick1/brick', 'srv2.local:/brick1/brick'],
```
Comment (igalic) [inline spec/classes/init_spec.rb]: o_<

does gluster even support just two bricks??
isn't that a guaranteed case for split-brain?

File: manifests/service.pp (hunk 2 above)
Comment (bastelfreak) [inline manifests/service.pp]: IIRC :: is recommended here

File: manifests/service.pp (hunk 1 above)
Comment (bastelfreak) [inline manifests/service.pp]: I failed. They should be there. I would add them back

File: spec/classes/init_spec.rb [hunk 5]
```diff
@@ -1,91 +1,106 @@
...
+            volumes: { 'data1' => %w(this is an array) }
           }
-        }
-      end
-      it 'should fail' do
-        expect {
-          should contain_gluster__volume('data1')
-        }.to raise_error(Puppet::Error, //)
+        end
+        it 'should fail' do
```
Comment (igalic) [inline spec/classes/init_spec.rb]: according to @jyaworski this titles will soon change with rspec-rubocop.

we can preemptively change it to `it 'fails'`

File: manifests/volume.pp (hunk 3 above)
Comment (bastelfreak) [inline manifests/volume.pp]: are you sure? we will hit the else part if it doesn't fail, then it is needed?

File: spec/classes/init_spec.rb (hunk 4 above)
Comment (bastelfreak) [inline spec/classes/init_spec.rb]: I think it works, but it probably isn't a good idea.

File: spec/classes/init_spec.rb (hunk 5 above)
Comment (bastelfreak) [inline spec/classes/init_spec.rb]: fixed in c80149346aa10e7e55c57633197af85c7f28aaf8

File: manifests/volume.pp (hunk 3 above)
Comment (igalic) [inline manifests/volume.pp]: what i mean is:
```puppet
   if $stripe {
     if ! is_integer( $stripe ) {
       fail("Stripe value ${stripe} is not an integer")
     }
     $_stripe = "stripe ${stripe}"
  } else {
…
```

Comment (bastelfreak) [inline manifests/volume.pp]: ahhh

Comment (bastelfreak) [inline manifests/volume.pp]: fixed in 4943144d72d6298a6722a0dab9a46f1929c31654

**Comment by jyaworski** [comment]
Squash?

**Comment by rnelson0** [comment]
FYI you no longer need :: prepends at all in puppet 4. With good design, it
shouldn't be an issue with puppet 3 either - mostly a case for profile
classes that match module names like profile::apache that includes apache.
But it would probably qualify as a backward incompatible change.

-- 

Rob Nelson
rnelson0@gmail.com

**Pull request merged**